Sources/ManageRegistration.php

```diff
@@ -292,16 +322,25 @@ function SetReserved()
 function ModifyRegistrationSettings($return_config = false)
 {
 	global $txt, $context, $scripturl, $modSettings, $sourcedir;
+	global $language, $boarddir;
 
 	// This is really quite wanting.
 	require_once($sourcedir . '/ManageServer.php');
 
+	// Do we have at least default versions of the agreement and privacy policy?
+	$agreement = file_exists($boarddir . '/agreement.' . $language . '.txt') || file_exists($boarddir . '/agreement.txt');
+	$policy = !empty($modSettings['policy_' . $language]);
+
 	$config_vars = array(
 		array('select', 'registration_method', array($txt['setting_registration_standard'], $txt['setting_registration_activate'], $txt['setting_registration_approval'], $txt['setting_registration_disabled'])),
 		array('check', 'send_welcomeEmail'),
-		'',
-
-		array('int', 'coppaAge', 'subtext' => $txt['zero_to_disable'], 'onchange' => 'checkCoppa();', 'onkeyup' => 'checkCoppa();'),
+	'',
+		array('check', 'requireAgreement', 'text_label' => $txt['admin_agreement'], 'value' => 1, 'disabled' => 1),
+		array('warning', empty($agreement) ? 'error_no_agreement' : ''),
+		array('check', 'requirePolicyAgreement', 'text_label' => $txt['admin_privacy_policy'], 'value' => 1, 'disabled' => 1),
+		array('warning', empty($policy) ? 'error_no_privacy_policy' : ''),
```

Comment:
why are these always disabled?

Reply:
Those values in SMF 2.0.16 were based on `!empty($modSettings['force_gdpr'])` but since that setting doesn't exist anymore, these values are being treated as if `force_gdpr` was always true

Reply:
Then you have it backwards. You can remove the disabled part I think.

Reply:
I think they wanted the agreement always required, but I might be wrong.

Reply:
@Sesquipedalian any input here?

Reply:
Both Agreements are forced in 2.x

Reply:
Therefore I think this conversation can be resolved then.